WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Container/AddProductWithAIContainerViewModel.swift

```diff
@@ -45,7 +51,9 @@ final class AddProductWithAIContainerViewModel: ObservableObject {
     }
 
     func onUsePackagePhoto(_ name: String?) {
-        // TODO: Launch UsePackagePhoto flow
+        presentPackageFlow? { data in
+            // TODO: Present preview
+        }
     }
```

Comment:
It seems to me that the event flow is quite complicated here: user taps packaging photo > container view triggers view model > view model triggers controller. 

Can we call directly from container view > view controller > view controller presents the package flow and then triggers a new method on the view model `didGenerateDataFromPackage(_ data: AddProductFromImageData)` on completion? That way, we can avoid an optional closure in the view model. How does that sound?

Reply:
Good point. I updated this in https://github.com/woocommerce/woocommerce-ios/pull/10738/commits/5f89ecb662f1bd3caf2b127ce4a36e93068a2c90